server/src/engine/socket.ts

```diff
@@ -85,60 +84,40 @@ export function isClosed() {
 let lastManualStopTimestamp: number = 0
 
 export function initialiseSocket({ uri, onOpen, onClose }: InitialiseSocketInput) {
-  if (!uri) {
-    throw 'Must be called with an uri'
-  }
-  webSocket = new WebSocket(uri)
+  webSocket = new ReconnectingWebSocket(uri, [], {
+    WebSocket,
+  })
 
-  webSocket.on('open', () => {
+  webSocket.addEventListener('open', function openHandler() {
     webSocketOpen = true
-    onOpen && setTimeout(onOpen!, 0)
+    if (onOpen !== undefined) {
+      setTimeout(onOpen, 0)
+    }
+
+    webSocket.removeEventListener('open', openHandler)
```

Comment:
Can you explain some of the above code, I am not sure how it works.

Reply:
I've tried improving it, but let me know if there's still something that should be explained better.